cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/PreferencesHttpHandler.java

```diff
@@ -55,12 +63,15 @@
 public class PreferencesHttpHandler extends AbstractAppFabricHttpHandler {
 
   private static final Gson GSON = new Gson();
-
+  private static final Gson DECODE_GSON = new GsonBuilder()
+      .registerTypeAdapterFactory(new CaseInsensitiveEnumTypeAdapterFactory())
+      .create();
   private final AccessEnforcer accessEnforcer;
   private final AuthenticationContext authenticationContext;
   private final PreferencesService preferencesService;
   private final Store store;
   private final NamespaceStore nsStore;
+  private JsonObject userRepo = new JsonObject();
```

Comment:
your repo is saved in memory ? In this case once server restarts repo info is lost. we should keep it in DB

Reply:
also try to add a new handler instead of putting your logic in PreferencesHttpHandler which is irrelavent to your logic.